core/store/src/trie/mod.rs

```diff
@@ -595,7 +719,12 @@ impl Trie {
         }
         let bytes = self.storage.retrieve_raw_bytes(hash)?;
         let node = RawTrieNodeWithSize::decode(&bytes).map_err(|err| {
-            StorageError::StorageInconsistentState(format!("Failed to decode node {hash}: {err}"))
+            StorageError::NotANode(NotANodeError {
+                hash: hash.clone(),
+                len: bytes.len(),
+                data_prefix: format!("{:?}", &bytes[..std::cmp::min(10, bytes.len())]),
```

Comment:
This screams to me SmallVec.  We might want to start using it.

Reply:
I would disagree here -- this is an error, and for errors you generally care about small size_of, rather than few allocations (allocations only affect sad path, size_of affects all paths). So what we want here is actually the oppostie -- to box this stuff away, such that the whole error is just a single pointer. 

But either way i expect the impact to be small.

Reply:
> this is an error, and for errors you generally care about small size_of, rather than few allocations

SmallVec capable of holding 16 bytes inline is the same size as String or Vec.  We can optimise allocations without needing to sacrifice size.

And if we want to optimise size than there’s also ArrayVec.